Backport of improvements in deployment from GF8 concurrency branch

dmatej opened the pull request:
Backported some cleanups from #24954 + much more rather small cleanups because originally it did not pass QL tests and I spent whole week investigating why it always breaks with every small change until I reached this safepoint.

* Added preInvoke before executing CDI extensions in WeldDeployer - some extensions need it.
* Added logging of ignored empty elements in SaxParserHandler
* Fixed few typos
* DeploymentException is now a standard exception - serialization of stacktraces was revisited in Java 9, so I expect that this is ok.
* RestExtension class was deleted, it is not required any more.
* ACLSingletonProvider - cleanup, revided comments, using ConcurrentHashMap instead of Hashtable
* Fixed NPE in EventListener (noticed once in logs)
* ComponentInvocation now has toString
* Added logs to WeldDeployer, - especially events are interesting
* CommonClassLoaderServiceImpl - removed attempts to load Derby from JDK (was possible in JDK6 which we don't support any more for a while)
* Replaced many `@Context` annotations with `@Inject`

### Note

* Disable whitespaces when reviewing, it will be more readable

dmatej committed:
Fixed warning - the error was not thrown

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed typo in ManagedExecutorDefinitionConverter's resource type

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Improved logging of SaxParserHandler, especially empty elements

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed ApplicationValidatorTest - changed message

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added toString to ComponentInvocation

- Useful in logs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed RestExtension - not required any more

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
WeldDeployer cleanup

- Using single lookup of InjectionManager
- The archiveFactory is private
- Removed dead try-finally
- Added space to failure messages.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Simplified and standardized DeploymentException

- The original code was probably required for some very old Java
- I this would cause some issues, just revert it.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
AMX cleanups + fixed missing log message keys

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Kernel cleanups - visibility, logging, removed DerbyInJDK support

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Minor cleanups - syntax, logging

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
CompositeUtil - added logging of processing RestExtension instances

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Admin REST Providers - part of cleanup

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Admin REST - another part of cleanups

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Anonymous class moved to own file - CdiBridge plus added logs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Admin REST adapters: API cleanup, renamed variables, added logs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed unused fields from RestService

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Admin REST, trivial improvements (generic maps, imports, etc)

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Admin REST: Replaced @Context with @Inject

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
The securityContext was not used

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Changing all @Context to @Inject except those using Ref

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
More logging in WeldDeployer

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed NPE in EventListener

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added invocationManager.preInvoke call to set component invocation info

- The invocation info might be required for weld extensions like Concurro which
  uses it in GF8.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Removed unused imports in CompositeUtil

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Fixed NPE in WebDeployer

- In recent merge I moved wrong curly brace when resolving conflicts

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
ACLSingletonProvider - improved comments

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

pzygielo commented on appserver/web/weld-integration/src/main/java/org/glassfish/weld/ACLSingletonProvider.java:
```diff
@@ -108,10 +103,27 @@ public T get(String id) {
+            LOG.log(DEBUG, () -> "clear(id=" + id + ")");
+            store.remove(getClassLoader());
+        }
+
```
A bit strange that `id`s are completely ignored in those methods. Could the parameters be renamed to express this intention?

dmatej replied:
Nope, the Singleton class is Weld's interface. The funny thing is that the id is somehow useful, however I doubt it is guaranteed that it is unique. Probably that is the reason why somebody uses the classloader:
`set(id=DeploymentDescriptorTests_web, object=org.jboss.weld.Container@3bf27b0)]]`

pzygielo replied:
>> Could the parameters be renamed to express this intention?

> Nope, the Singleton class is Weld's interface. 

But Weld does not have to know what is done to the parameter name in this implementation.
I find it also a little disturbing that it's logged like `id` values matter, while they don't.

dmatej replied:
So what is your idea? I don't like it too, it smells, I see in logs it is probably always the application name and maybe it could be used as a key instead of classloader, but ... I can 

* delete all these logs (but then I cannot read anything useful in logs)
* change `id=` to `application=` (but that can change one day without warning, it is rather guessing what it really is)
* switch from CL to the id (but that would be quite risky experiment now)
* keep it with grinding teeth (confusing as you wrote)
* add `", but using thread classloader as an id instead"`

I did this because it was the simplest way to get at least something.

pzygielo replied:
Do not consider my comment to be blocking :) (But I've not reviewed everything yet.).

I was only thinking about change `id` -> `ignoredId` - and keep everything else as-is. Just for me, reading this somewhere in the future.

I'm for marking this one detail as `resolved`, and leave it on _to think about list_.

dmatej replied:
Finally I added at least the comment to the log.

dmatej commented:
Don't merge it yet, I will yet need to extend the preInvoke-postInvoke range - I rebased the #24954  to this locally and it failed few tests with qualifiers which was possible to fix just by this change. However I still need to find out how large range is required.

dmatej committed:
Extended preInvoke-postInvoke scope

- Tested on GF8 where it is required for use of classifiers in Concurrency TCK

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
WeldDeployer refactoring

- Split processing events - long method with parts with clear borders.
- Logging enabling/disabling CDI
- ModuleInfo.toString usable in logs

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

avpinchuk commented on appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/DOLUtils.java:
```diff
@@ -370,12 +370,12 @@ public static ArchiveType getModuleType(String moduleType) {
     // WLS DD when both present in the same archive
     public static boolean isGFDDOverWLSDD() {
-        return Boolean.valueOf(System.getProperty(GFDD_OVER_WLSDD));
+        return Boolean.parseBoolean(System.getProperty(GFDD_OVER_WLSDD));
```
Why not use `Boolean.getBoolean()` instead in all places?

dmatej committed:
Moved "disabling Weld" log so it would appear just once in logs

- Because event() reacts on three kinds of events it appeared three times

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Added note to logs that we don't use the id to find the ACL Singleton.

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Using getBoolean in DolUtils

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

dmatej committed:
Lower logging level in CdiBridge - forgotten in the same PR

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>

avpinchuk commented on nucleus/common/internal-api/src/main/java/org/glassfish/internal/data/ApplicationInfo.java:
```diff
@@ -112,12 +112,12 @@ private void createServiceLocator() {
         appServiceLocator = slf.create(locatorName);
         deploymentFailedListener = new DeploymentFailedListener(source);
-        events.register(deploymentFailedListener);
+        unregisterEventListener(deploymentFailedListener);
```
Maybe we should **register** listener here instead of *unregister*?

dmatej replied:
Do you have any idea why it did not break any test? Whole TCK passed, I am just locally running just few remaining which failed because of read timeouts caused by my slow network.
That is really stupid mistake I made. I do wonder how I did that ... thank you for noticing that!

avpinchuk replied:
`DeploymentFailedListener` calls `disposeServiceLocator()` method. This method unregister the listener itself and destroy the application service locator in case of a deployment failure. I think, we just have a memory leak in this case.

dmatej committed:
Fixed typo made in the same PR - registerEventListener

Signed-off-by: David Matějček <david.matejcek@omnifish.ee>
